proxy/proxy.go

```diff
@@ -723,11 +724,11 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 				}
 			}
 		}
-		p.log.Errorf("error during backend roundtrip: %s: %v", ctx.route.Id, err)
 
 		if cerr := req.Context().Err(); cerr != nil {
-			p.log.Errorf("Failed to do request, because of context: %v", cerr)
-			return nil, &proxyError{err: cerr, code: http.StatusGatewayTimeout}
+			// deadline exceeded or canceled in stdlib, client closed request
+			// see https://github.com/zalando/skipper/issues/687#issuecomment-405557503
+			return nil, &proxyError{err: cerr, code: 499}
```

Comment:
if this is a request that the client has already cancelled, then this status code is used only for logging. Is this correct? If yes, then i'd prefer leaving the status code 0, because it is something that we use only for logging.

Reply:
It's also added to the metrics, so that's why a 5xx is a problem because it gives the wrong impression of the service error rate.

Counting this as a 4xx seems right because we can exclude it from errors caused by the proxy/backend, but we can still monitor the 4xx rate and be aware of the error rate.

It would be interesting to know what AWS ELBs report in this case.

Reply:
Do you think the AWS ELB is better for the status code handling then nginx?
I mean AWS ELB also have a very different way of handling X-For-* headers and set the clientip last instead of all others that put this first as shown in an example of the RFC.

Reply:
> Do you think the AWS ELB is better for the status code handling then nginx?

AWS ELB is just interesting because this is what our users are used to.

> For me the biggest drawback to status code 0 is that:

Note that status code `0` will be reported as `500` in logs and metrics. This is my biggest concern because it will give the wrong impression to users.

I agree with the benefit of `499` being fairly simple to search for.

Reply:
i got convinced, thanks for explaining it. Let's use 499.